modality-probe-debug-collector/src/cli.rs

```diff
@@ -0,0 +1,618 @@
+use std::error::Error;
+use std::fmt;
+use std::fs::File;
+use std::io::prelude::*;
+use std::net::SocketAddrV4;
+use std::path::PathBuf;
+
+use parse_duration::parse;
+use structopt::StructOpt;
+
+use goblin::container::Endian;
+use goblin::elf::Elf;
+
+use modality_probe_debug_collector::{Config, ProbeAddr};
+
+/// Error in options processing
+#[derive(Debug)]
+struct OptionsError {
+    details: String,
+}
+
+impl OptionsError {
+    fn new(msg: &str) -> Self {
+        OptionsError {
+            details: msg.to_string(),
+        }
+    }
+}
+
+impl fmt::Display for OptionsError {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "{}", self.details)
+    }
+}
+
+impl Error for OptionsError {
+    fn description(&self) -> &str {
+        &self.details
+    }
+}
+
+#[derive(Debug, Default, StructOpt)]
+#[structopt(
+    name = "modality-probe-debug-collector",
+    about = "Periodically collects logs from microcontrollers over debug interfaces; outputs them to a file."
+)]
+pub struct CLIOptions {
+    /// Session id to associate with the collected trace data
+    #[structopt(short = "s", long = "session-id", default_value = "0")]
+    session_id: u32,
+
+    /// Specifies little endian architecture of target system
+    #[structopt(short = "l", long = "little-endian")]
```

Comment:
I think you can add another config option here to lean more on StructOpt: `conflicts_with = "big-endian"`